src/vector/platform/ElectronPlatform.js

```diff
@@ -129,6 +131,40 @@ export default class ElectronPlatform extends VectorBasePlatform {
         ipcRenderer.send('loudNotification');
     }
 
+    initSpellCheckHandler(lang) {
+        window.spellCheckHandler = new SpellCheckHandler();
+        window.spellCheckHandler.attachToInput();
+
+        window.spellCheckHandler.switchLanguage(lang);
+        window.spellCheckHandler.autoUnloadDictionariesOnBlur();
+        let contextMenuBuilder = new ContextMenuBuilder(window.spellCheckHandler);
+        const seperatorItem = [{ type: 'separator' }];
+
+        async function generateMenu(contextInfo, MainMenu) {
+            let SpellMenu = new Menu();
+            await contextMenuBuilder.addSpellingItems(SpellMenu, contextInfo);
+            contextMenuBuilder.addSearchItems(SpellMenu, contextInfo);
+
+            const newMainMenuTemplate = [...MainMenu.items, ...seperatorItem, ...SpellMenu.items];
+            const newMainMenu = Menu.buildFromTemplate(newMainMenuTemplate);
+
+            newMainMenu.popup(remote.getCurrentWindow(), { async: true });
+        }
+
+        async function showMenu(contextInfo) {
+            ipcRenderer.send('SelectedContextMenu', contextInfo);
+            ipcRenderer.once('SelectedContextMenu-reply', async function(event, arg) {
+                await generateMenu(contextInfo, arg);
+            })
+        }
+
+        let contextMenuListener = new ContextMenuListener(async function(info) {
+            await showMenu(info);
```

Comment:
not sure why this is an async function, its essentially just a normal fn which does `return showMenu(info)`